packages/aws-cdk/toolkit/lib/cli-io-host.ts

```diff
@@ -0,0 +1,101 @@
+import * as chalk from 'chalk';
+
+/**
+ * Basic message structure for toolkit notifications.
+ * Messages are emitted by the toolkit and handled by the IoHost.
+ */
+interface IoMessage {
+  time: Date;
+  level: IoMessageLevel;
+  action: IoAction;
+  code: string;
+  message: string;
+  forceStdout?: boolean;
+  // Optionally specify Chalk style for stdout/stderr, if TTY is enabled
+  style?: ((str: string) => string);
+}
+
+export type IoMessageLevel = 'error' | 'warn' | 'info' | 'debug' | 'trace';
+
+export type IoAction = 'synth' | 'list' | 'deploy' | 'destroy';
+
+interface CliIoHostOptions {
```

Comment:
same comment about documentation. it's important to easily figure out what tty / ci are and why they matter when it comes to logging

Reply:
added some explanation here and on getStream